netsim/cli/connect.py

```diff
@@ -103,7 +103,9 @@ def ssh_connect(
   if data.netlab_ssh_args:
     c_args.extend(data.netlab_ssh_args.split(' '))
 
-  if data.ansible_ssh_pass:
+  if data.ansible_ssh_private_key_file:
+    c_args.extend(['-i', strings.eval_format(data.ansible_ssh_private_key_file,{'name': data.host})])
```

Comment:
What's wrong with passing the whole node data, and if you need the 'name' parameter, you can adjust the 'adjust_inventory_host' call to retain it. This is yet another example of the "let's fix my immediate problem" mentality.